R/data_color.R

```diff
@@ -391,131 +393,118 @@ adjust_luminance <- function(colors,
   # Calculate new luminance values based on a fixed step-change in `x`
   y_2 <- 1 / (1 + exp(-(x + steps)))
 
-  # Rescale new luminance values to [0, 100]
+  # Rescale the new luminance values to [0, 100]
   l <- y_2 * 100.
 
   # Obtain hexadecimal colors from the modified HCL color values
-  hcl_colors <- grDevices::hcl(h, c, l, alpha = NULL)
-
-  # Apply alpha values to the hexadecimal colors
-  hcl_colors <- paste0(hcl_colors, alpha)
-
-  hcl_colors
+  grDevices::hcl(h, c, l, alpha = alpha)
 }
 
-is_hex_col_w_alpha <- function(color) {
-  color %>% tidy_grepl("^#[0-9A-Fa-f]{8}$")
-}
 
-is_hex_col_no_alpha <- function(color) {
-  color %>% tidy_grepl("^#[0-9A-Fa-f]{6}$")
-}
-
-frac_alpha_to_hex <- function(alpha) {
-  grDevices::rgb(0, 0, 0, alpha) %>% substr(8, 9)
-}
-
-#' Extract a vector of alpha values for a vector of colors
+#' Is a string a color value in rgba() format?
+#'
+#' The input for this is a character vector of length 1. The output is a single
```

Comment:
```suggestion
#' The input for this is a character vector of any length. The output is a vector of
```

Reply:
I've completely rewritten the docs for this internal function to clearly explain what how it works and what it's used for. It is vectorized.